pkg/middlewares/aws_iam_role_defaults.go

```diff
@@ -0,0 +1,67 @@
+package middlewares
+
+import (
+	"github.com/sirupsen/logrus"
+
+	"github.com/cloudskiff/driftctl/pkg/resource"
+	"github.com/cloudskiff/driftctl/pkg/resource/aws"
+)
+
+// When scanning a brand new AWS account, some users may see irrelevant results about default AWS role policies.
+// We ignore these resources by default when strict mode is disabled.
+type AwsIamRoleDefaults struct{}
+
+var ignoredIamRoleIds = []string{
+	"AWSServiceRoleForSSO",
+	"OrganizationAccountAccessRole",
+}
+
+func NewAwsIamRoleDefaults() AwsIamRoleDefaults {
+	return AwsIamRoleDefaults{}
+}
+
+func (m AwsIamRoleDefaults) Execute(remoteResources, resourcesFromState *[]resource.Resource) error {
+	newRemoteResources := make([]resource.Resource, 0)
+
+	for _, remoteResource := range *remoteResources {
+		// Ignore all resources other than iam role
+		if remoteResource.TerraformType() != aws.AwsIamRoleResourceType {
+			newRemoteResources = append(newRemoteResources, remoteResource)
+			continue
+		}
+
+		existInState := false
+		for _, stateResource := range *resourcesFromState {
+			if resource.IsSameResource(remoteResource, stateResource) {
+				existInState = true
+				break
+			}
+		}
+
+		if existInState {
+			newRemoteResources = append(newRemoteResources, remoteResource)
+			continue
+		}
+
+		isIgnored := false
+		for _, id := range ignoredIamRoleIds {
+			if remoteResource.TerraformId() == id {
```

Comment:
Actually they can be many of `AWSServiceRole*` roles.

![image](https://user-images.githubusercontent.com/6154987/113273392-e1948700-92dc-11eb-9888-d50d5f37a272.png)

I think for roles we should ignore them using the path attribute

![image](https://user-images.githubusercontent.com/6154987/113273512-025cdc80-92dd-11eb-96f7-ec8c2b9b6457.png)

Every role with a path starting with `/aws-service-role` should be skipped in strict mode **if they are not managed by IaC**

Let's ask our AWS certified guy @wbeuil about it, is it the right logic ?

Reply:
This is what a service-linked role is:

`A service-linked role is a unique type of IAM role that is linked directly to an AWS service. Service-linked roles are predefined by the service and include all the permissions that the service requires to call other AWS services on your behalf. The linked service also defines how you create, modify, and delete a service-linked role. A service might automatically create or delete the role. It might allow you to create, modify, or delete the role as part of a wizard or process in the service. Or it might require that you use IAM to create or delete the role. Regardless of the method, service-linked roles make setting up a service easier because you don't have to manually add the necessary permissions for the service to complete actions on your behalf.`

We can easily ignore them by default